collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java

```diff
@@ -45,17 +47,15 @@
 import static org.odk.collect.android.preferences.GeneralKeys.CONSTRAINT_BEHAVIOR_ON_SWIPE;
 import static org.odk.collect.android.preferences.PreferencesActivity.INTENT_KEY_ADMIN_MODE;
 
-public class AdminPreferencesFragment extends BasePreferenceFragment implements Preference.OnPreferenceClickListener {
+public class AdminPreferencesFragment extends PreferenceFragmentCompat implements Preference.OnPreferenceClickListener {
 
     public static final String ADMIN_PREFERENCES = "admin_prefs";
 
     @Override
     public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
-        super.onCreatePreferences(savedInstanceState, rootKey);
-
         getPreferenceManager().setSharedPreferencesName(ADMIN_PREFERENCES);
-
         setPreferencesFromResource(R.xml.admin_preferences, rootKey);
+        ((CollectAbstractActivity) getActivity()).initToolbar(getPreferenceScreen().getTitle());
```

Comment:
Can we do a conditional cast here like in `BasePreferencesFragment`?

Reply:
Now it doesn't exist there and we set title only in BasePreferenceFragment which works for every screen.